var/spack/repos/builtin/packages/root/package.py

```diff
@@ -791,18 +803,21 @@ def setup_dependent_build_environment(
         env.append_path("CMAKE_MODULE_PATH", self.prefix.cmake)
         env.prepend_path("ROOT_INCLUDE_PATH", dependent_spec.prefix.include)
         if "+rpath" not in self.spec:
-            env.prepend_path("LD_LIBRARY_PATH", self.prefix.lib.root)
+            env.prepend_path(self.root_library_path, self.prefix.lib.root)
         if "platform=darwin" in self.spec:
             # Newer deployment targets cause fatal errors in rootcling
             env.unset("MACOSX_DEPLOYMENT_TARGET")
 
     def setup_dependent_run_environment(
         self, env: spack.util.environment.EnvironmentModifications, dependent_spec
     ):
-        env.set("ROOTSYS", self.prefix)
-        env.set("ROOT_VERSION", "v{0}".format(self.version.up_to(1)))
```

Comment:
These additions were made per https://github.com/spack/spack/pull/13657#discussion_r345463706. Are we sure we don't introduce regressions by removing them from the dependent run environment?

I guess your reasoning is maybe that `setup_dependent_run_environment` will already require `setup_run_environment`, but that would only be true if `dependent_spec` depends on `root` with `type=("run",...)`...

I don't disagree here, but I would just want to make sure we don't have regressions.

Reply:
The [code](https://github.com/spack/spack/blob/8128b549a54b176e4dcaa953b63ee43f1428934c/lib/spack/spack/build_environment.py#L1052,L1055) for invoking `setup_dependent_run_environment` looks like:

```python
for spec in dspec.dependents(deptype=dt.LINK | dt.RUN):
    if id(spec) in self.nodes_in_subdag:
        pkg.setup_dependent_run_environment(run_env_mods, spec)
pkg.setup_run_environment(run_env_mods)
```

which means that `setup_run_environment` is always invoked whenever `setup_dependent_run_environment` is invoked (but not necessarily vice versa).   Because `ROOTSYS` and `ROOT_VERSION` will be set by `setup_run_environment` I do not believe we will encounter a regression.